tests/comments/flow_union.js

```diff
@@ -0,0 +1,9 @@
+type UploadState<E, EM, D>
+  // The upload hasnt begun yet
```

Comment:
I didn't add any code to handle this case (and wasn't going to). I just implemented the fix for your example and as a happy coincidence it also fixed this one.